src/query_plan/fetch_dependency_graph.rs

```diff
@@ -2202,27 +2251,155 @@ fn compute_nodes_for_op_path_element<'a>(
     Ok(updated)
 }
 
-fn wrap_input_selections(
-    _wrapping_type: &CompositeTypeDefinitionPosition,
-    _selections: &NormalizedSelectionSet,
-    _context: &OpGraphPathContext,
-) -> NormalizedSelectionSet {
-    todo!() // Port `wrapInputsSelections` in `buildPlan.ts`
+/// A helper function to wrap the `initial` value with nested conditions from `context`.
+fn wrap_selection_with_type_and_conditions<T>(
+    supergraph_schema: &ValidFederationSchema,
+    wrapping_type: &CompositeTypeDefinitionPosition,
+    context: &OpGraphPathContext,
+    initial: T,
+    wrap_in_fragment: fn(NormalizedInlineFragment, T) -> T,
```

Comment:
I went with `fn`, since it was just enough and the function wasn't really reusable outside of the current usage. Having said that. `FnMut` would be more future-proof. Thanks for the detailed explanation!

Reply:
Another difference is that the compiler can more easily inline an `impl Fn*` than a function pointer